tests/expectations/metrics/test_core.py

```diff
@@ -6964,3 +7029,84 @@ def test_map_select_column_values_unique_within_record_spark(spark_session):
         {"a": 1.0, "b": 1.0, "c": 2.0},
         {"a": 4.0, "b": 4.0, "c": 4.0},
     ]
+
+
+def test_dx_524_explicit_string_identifiers_should_work_with_validator(spark_session):
```

Comment:
I dont think we use this convention (dx-524) for naming tests, but I see the benefit. Another option could be to include it in a docstring, but I think either way could work.

Reply:
Sounds good - I can move it to a docstring or a comment

Reply:
How would the external reader be able to interpret "dx_524"?

Reply:
Ah true! I will mention the external github issue instead